rewrite-test/src/main/kotlin/org/openrewrite/java/typegeneration/TypeGenerationTest.kt

```diff
@@ -0,0 +1,512 @@
+//            rewriteRoot.resolve("rewrite-core/src/main/java/org/openrewrite/config/DeclarativeNamedStyles.java"),
+//            rewriteRoot.resolve("rewrite-core/src/main/java/org/openrewrite/config/DeclarativeRecipe.java"),
+
+    /**
+     * This method will use load the source file represented by the path and then parse that file with the parser. It
+     * will then extract the types out of the generates AST and then recompile the source with a new parser (that does
+     * not have the classpath of the original parser) that is seeded with just the types extracted from the AST. The
+     * compilation unit from the second parse is then passed into the afterCondtions method and that method can be
+     * used to verify type attribution was successful.
+     */
+    fun assertCompilationWithByteCode(
+        path: Path,
+        parser: JavaParser = this.parser,
+        checkCompilationUnit: (J.CompilationUnit) -> Unit = {},
+        checkCompilerOutput: (String) -> Unit = this.defaultCompilerOutputHandler,
+    ) {
+        val inputs = Arrays.asList(path)
+        val standardOut = System.out
+        val outputStreamCaptor = ByteArrayOutputStream()
+
+        PrintStream(outputStreamCaptor).use {
+            try {
+                System.setOut(PrintStream(outputStreamCaptor))
+                println("------------------- Initial Compilation (${path}) --------------------------------------")
+                var compilationUnit = parser.parse(inputs, null, InMemoryExecutionContext())[0]
+                val typeInformation = TypeInformation()
+                TypeExtractionVisitor().visit(compilationUnit, typeInformation)
+                val dynamicClasses = typeInformation.getTypesAsByteCode()
+
+                println("------------------- Byte Code Only Compilation (${path}) -------------------------------")
+                compilationUnit = JavaParser.fromJavaVersion()
+                    .logCompilationWarningsAndErrors(true)
+                    .classpath("")
+                    .classpath(*dynamicClasses.toTypedArray())
+                    .build()
+                    .parse(inputs, null, InMemoryExecutionContext())[0]
+
+                System.setOut(standardOut)
+                checkCompilerOutput(outputStreamCaptor.toString())
+                checkCompilationUnit(compilationUnit)
+            } finally {
+                System.setOut(standardOut)
+            }
+
+        }
+    }
+
+    /**
+     * This method will parse the source string using the parser that is passed in. It will then extract the types out
+     * of the generated AST and then recompile the source with a new parser (that does not have the classpath of the
+     * original parser) that is seeded with just the types extracted from the original AST.The
+     * compilation unit from the second parse is then passed into the afterCondtions method and that method can be
```

Comment:
"afterConditions"? There's no parameter called that. I imagine it got renamed to "checkCompilationUnit".